packages/std/src/traits.rs

```diff
@@ -83,10 +88,28 @@ pub trait Querier {
     /// types. People using the querier probably want one of the simpler auto-generated
     /// helper methods
     fn raw_query(&self, bin_request: &[u8]) -> QuerierResult;
+}
+
+#[derive(Copy, Clone)]
+pub struct QuerierWrapper<'a> {
+    querier: &'a dyn Querier,
+}
```

Comment:
Why did you create this extra type?

Reply:
To solve a headache.

The `Querier` trait was implementing a lot of automatic functions (beyond the one we implement). These are generic functions. `&dyn Querier` as a trait objects does not support traits with generic functions (as there is not a fixed vtable with all possible trait functions). You can look in the history, there was a place where I pulled out S and A to &dyn Storage and &dyn Api, but left Querier. I had to add this type to get Querier to compile.

All those Querier "helper functions" can no longer be on the trait. Wrapping it in a struct seemed the easiest way to add it. But I guess using a newtype `pub struct QuerierWrapper<'a>(&'a dyn Querer)` is actually a clearer representation of how we wrap - like `Binary`/`CanonicalAddr` wrap `Vec<u8>`.

Secondly, we can adjust naming. Maybe `QuerierWrapper`=>`Querier` (as this is what contract devs will be seeing), and `Querier` => `QuerierTrait` or such, which is only used internally by core devs. Same way to make `Deps`, `DepsRef` and simplify the naming in the public API by giving the internals a more clunky naming.

Reply:
I did the rename in `a7efaa3`. Happy to revert it if you don't like it, but I think this solves the issue with a cleaner public API.